### PR TITLE
Dahl buttons

### PR DESCRIPTION
**Pull request by pswk1**
button migration for Dahl

### CODE REVIEW

File: src/components/Authentication/LoginPage.js [hunk 1]
```diff
@@ -14,6 +14,8 @@ import Typography from "@material-ui/core/Typography";
 import InputAdornment from "@material-ui/core/InputAdornment";
 import EmailOutlinedIcon from "@material-ui/icons/EmailOutlined";
 
+import { ResponsiveButton } from '../../theme/ThemedComponents/Button/ResponsiveButton';
```
Comment (andrewjknapp) [inline src/components/Authentication/LoginPage.js:17]: Some of your import statement use relative directories ../../theme others just start at theme/ We should keep this consistent throughout and ask Daniel which would be preferred

File: src/components/Authentication/LoginPage.js [hunk 2]
```diff
@@ -152,21 +155,52 @@ const LoginPage = () => {
...
                                             email,
                                             password,
                                         },
-                                    }} variant="outlined">
+                                    }} 
+                                    variant="outlined"
+                                >
                                     CREATE ACCOUNT
-                                </Button>
+                                </ResponsiveButton>
```
Comment (andrewjknapp) [inline src/components/Authentication/LoginPage.js:171]: Be sure to remove these commented out buttons

File: src/components/Authentication/LoginPage.js [hunk 3]
```diff
@@ -152,21 +155,52 @@ const LoginPage = () => {
...
-                                <Button data-cy="nextButton" className="signInButton" onClick={handleCheck}>
+                                <ResponsiveButton 
+                                    data-cy="nextButton" 
+                                    className="signInButton" 
+                                    onClick={handleCheck}
+                                    variant='outlined'
+                                >
                                     SIGN IN
-                                </Button>
+                                </ResponsiveButton>
```
Comment (andrewjknapp) [inline src/components/Authentication/LoginPage.js:195]: Same here with the commented out buttons

File: src/components/FeatureViews/Accounts/Accounts.js [hunk 4]
```diff
@@ -292,20 +294,26 @@ const Accounts = () => {
...
                         <Grid container item md={3}>
                             <Grid className="toggleView" item md={6}>
-                                <Button
-                                    className={`btn list ${viewToggle && "active"}`}
-                                    onClick={setView(true)}>
-                                    <ListView className={`icon ${viewToggle && "active"}`} />
-                                List View
-                            </Button>
+                                <ResponsiveButton
+                                    // className={`btn list ${viewToggle && "active"}`}
```
Comment (andrewjknapp) [inline src/components/FeatureViews/Accounts/Accounts.js:298]: Why are the className and startIcon commented out in these buttons on line 297 and 308?

File: src/components/FeatureViews/Accounts/TabComponents/EnrollmentView.js [hunk 5]
```diff
@@ -342,9 +343,9 @@ const CourseSessionStatus = () => {
                                 parentOfCurrentStudent={studentParent} />
                         </Grid>
                         <Grid item>
-                            <Button className="button unenroll" onClick={openUnenrollDialog}>
+                            <ResponsiveButton Button className="button unenroll" onClick={openUnenrollDialog}>
```
Comment (andrewjknapp) [inline src/components/FeatureViews/Accounts/TabComponents/EnrollmentView.js:346]: Might want to put these attributes on their own lines, it is getting a bit long

File: src/components/FeatureViews/AdminPortal/AdminActionCenter.js [hunk 6]
```diff
@@ -73,38 +75,41 @@ const AdminActionCenter = () => {
...
-				<Button
+				<ResponsiveButton
 					aria-controls="simple-menu"
 					aria-haspopup="true"
 					className={`button ${tabState.user && "active"}`}
 					onClick={handleClick(setUserAnchor)}
+					variant='outlined'
 				>
-					<NewInstructor className="admin-action-icon" />
+					{/* <NewInstructor className="admin-action-icon" /> */}
```
Comment (andrewjknapp) [inline src/components/FeatureViews/AdminPortal/AdminActionCenter.js:85]: Is this line not needed? If so get rid of it. also lines 97 and 110 have similar comments

File: src/components/FeatureViews/Availability/CreateOOOForm.js [hunk 7]
```diff
@@ -194,38 +195,62 @@ export default function CreateOOOForm() {
...
-                                onClick={() =>
-                                    SubmitNoticeChild.current.handleClearForm()
-                                }
-                                color="primary"
-                                variant="outlined"
-                                data-cy="clear-OOO-button"
-                            >
-                                Clear
-                            </Button>
+                            <ResponsiveButton
```
Comment (andrewjknapp) [inline src/components/FeatureViews/Availability/CreateOOOForm.js:198]: Be sure to get rid of these comments in this responsiveButton as well as starting at line 214

File: src/components/FeatureViews/Availability/CreateOOOForm.js [hunk 8]
```diff
@@ -194,38 +195,62 @@ export default function CreateOOOForm() {
...
-                                color="primary"
+                                // color="primary"
                                 variant="outlined"
                                 data-cy="submit-OOO-button"
                             >
-                                Submit
-                            </Button>
+                                submit
+                            </ResponsiveButton>
+                            // <Button
```
Comment (andrewjknapp) [inline src/components/FeatureViews/Availability/CreateOOOForm.js:241]: No reason to save the old button,
"Let the past die! Kill it if you have to!"

File: src/components/FeatureViews/Courses/Announcements.js [hunk 9]
```diff
@@ -30,11 +35,11 @@ const useStyles = makeStyles({
   },
   newNoteButton: {
     marginBottom: "2em",
-    border: "1px solid #999999",
-    borderRadius: "5px",
-    fontSize: ".75rem",
-    fontWeight: 300,
-    fontFamily: "Roboto",
+    // border: "1px solid #999999",
```
Comment (andrewjknapp) [inline src/components/FeatureViews/Courses/Announcements.js:38]: Get these comments out of here!

File: src/components/FeatureViews/Courses/ClassSessionView.js [hunk 10]
```diff
@@ -49,11 +51,11 @@ const useStyles = makeStyles((theme) => ({
   },
   newNoteButton: {
     marginBottom: "2em",
-    border: "1px solid #999999",
-    borderRadius: "5px",
-    fontSize: ".75rem",
-    fontWeight: 300,
-    fontFamily: "Roboto",
+    // border: "1px solid #999999",
```
Comment (andrewjknapp) [inline src/components/FeatureViews/Courses/ClassSessionView.js:54]: Here is another one

File: src/components/FeatureViews/Registration/RegistrationActions.js [hunk 11]
```diff
@@ -101,40 +103,47 @@ const RegistrationActions = () => {
 				<Grid item md={9}>
 					{(currentParent || parentIsLoggedIn) && (
 						<Grid item xs={2}>
-							<Button
+							<ResponsiveButton 
 								aria-controls="simple-menu"
 								aria-haspopup="true"
-								className="button"
+								// className="button"
```
Comment (andrewjknapp) [inline src/components/FeatureViews/Registration/RegistrationActions.js:109]: If we don't need the className make it gone

File: src/components/FeatureViews/Registration/RegistrationActions.js [hunk 12]
```diff
@@ -101,40 +103,47 @@ const RegistrationActions = () => {
...
 				<Grid item xs={2}>
 					{registeringParent ? (
 						!data && <Tooltip title="Registering Parent">
-							<Button className="button" onClick={openDialog}>
-								<div
+							<ResponsiveButton 
+								variant='contained'
+								// className="button" 
+								onClick={openDialog}>
+								{/* <div
```
Comment (andrewjknapp) [inline src/components/FeatureViews/Registration/RegistrationActions.js:127]: If we don't need this icon you can delete it

File: src/components/FeatureViews/Registration/RegistrationActions.js [hunk 13]
```diff
@@ -101,40 +103,47 @@ const RegistrationActions = () => {
...
 					) : (
-						<Button className="button set-parent" onClick={openDialog} data-cy="select-parent">
-							<div className="circle-icon"/>
+						<ResponsiveButton 
+							variant='contained'
+							// className="button set-parent" 
+							onClick={openDialog} 
+							data-cy="select-parent"
+						>
+							{/* <div className="circle-icon"/> */}
```
Comment (andrewjknapp) [inline src/components/FeatureViews/Registration/RegistrationActions.js:144]: Comment

File: src/components/FeatureViews/Registration/RegistrationCartPayment.js [hunk 14]
```diff
@@ -124,36 +124,46 @@ const Payment = ({
...
 									className="button"
+									variant='contained'
 									disabled={!allValid}
 									onClick={updateQuantity}
-								>
-									UPDATE SESSIONS
-								</Button>
+									label='update sessions'
+								/>
+								// <Button
```
Comment (andrewjknapp) [inline src/components/FeatureViews/Registration/RegistrationCartPayment.js:160]: Here is another old button

File: src/components/OmouComponents/BackButton.js [hunk 15]
```diff
@@ -68,7 +69,9 @@ const BackButton = (props) => {
 
 	const renderIcon = () => {
 		if (props.icon == null) {
-			return <BackArrow fontSize="small" className="icon" />
+			return <BackArrow style={{transform: "scale(0.8)"}}/>
+			//<BackArrow fontSize="small" className="icon" />
```
Comment (andrewjknapp) [inline src/components/OmouComponents/BackButton.js:73]: comment

File: src/components/OmouComponents/BackButton.js [hunk 16]
```diff
@@ -77,15 +80,22 @@ const BackButton = (props) => {
...
 		<Hidden mdDown>
-			<Button
+			<ResponsiveButton 
+				variant='outlined'
+				className="control course button"
+				onClick={() => { handleClick() }}
+			>
+				{label()}
+			</ResponsiveButton>
+			{/* <Button
```
Comment (andrewjknapp) [inline src/components/OmouComponents/BackButton.js:90]: Get rid of these old buttons

File: src/theme/ThemedComponents/Button/ResponsiveButton.js [hunk 17]
```diff
@@ -2,40 +2,59 @@ import React from 'react'
...
+           buttonWidth = 160;
        }
-       return buttonWidth;
+       return buttonWidth;;
    };
 
-/* To utilize an icon in this button, import the icon in your component and pass it in through the startIcon prop to have it 
-on the left of the label and endIcon to have it on the right. Example: startIcon={<AddIcon />}
+/* 
+Using icons:
```
Comment (andrewjknapp) [inline src/theme/ThemedComponents/Button/ResponsiveButton.js:21]: Good documentation, I am going to have to do this with mine

**Review by andrewjknapp — CHANGES_REQUESTED** [review]
Things to look out for
1) Fix your import statements so they all start with theme/ or they are all relative with ../
2) Delete your comments. I marked all the places that I found them so go through and get those.

File: src/components/FeatureViews/Availability/CreateOOOForm.js (hunk 8 above)
Comment (pswk1) [reply src/components/FeatureViews/Availability/CreateOOOForm.js:241]: TIL I am Rey

**Pull request merged**